### PR TITLE
Add `zcutil/clean.sh` and `zcutil/distclean.sh` scripts that work reliably (unlike `make clean`)

### PR DESCRIPTION
**Pull request by daira**
Workaround for #2226.

### CODE REVIEW

**Comment by rex4539** [comment]
From my understanding, this more like `make distclean` than `make clean`.

**Comment by str4d** [comment]
[Yep](https://www.gnu.org/software/automake/manual/automake.html#Clean). Please name this `zcutil/distclean.sh` so it matches user expectations.

**Comment by daira** [comment]
My understanding is that it's a matter of opinion whether `clean` should delete files that were created by `./configure`. I think it should. In any case, I've split the script into `clean.sh` which doesn't delete built dependencies, and `distclean.sh` which does.

File: zcutil/distclean.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+zcutil/clean.sh
+
+rm -rf depends/*-*-*
+rm -rf depends/built
+rm -rf depends/sources
+rm -rf target
+
+rm -f .cargo/config
+rm -f .cargo/.configured-for-offline
```
Comment (str4d) [inline zcutil/distclean.sh]: - These are generated by `src/Makefile.am`, so should be removed by `zcutil/clean.sh`
- Missing `.cargo/.configured-for-online`

**Comment by str4d** [comment]
Original PR comment (removed so it's not in the merge commit):

---

refs #2226 

To test this,
* start from a fresh clone of `zcash/zcash`;
* build as usual with `zcutil/build.sh`;
* run `zcutil/distclean.sh`;
* run `zcutil/distclean.sh` again (to check that it's idempotent and doesn't depend on the files existing);
* build again;
* run `zcutil/distclean.sh`;
* run the following commands, which list any files outside the `.git` directory that are not checked in:
  ```
  git ls-files |sort >../lsfiles
  find -type f |sed 's|^[.]/||' |grep -v '^.git/' |sort >../findfiles
  grep -Fxvf ../lsfiles ../findfiles
  ```
  The output should be empty.
* build again;
* run `zcutil/clean.sh`;
* run the following commands, which list any files outside the `.git`, `depends`, `target`, and `.cargo` directories that are not checked in:
  ```
  git ls-files |sort >../lsfiles
  find -type f |sed 's|^[.]/||' |grep -v '^.git/' |grep -v '^depends/' |grep -v '^target/' |grep -v '^.cargo/' |sort >../findfiles
  grep -Fxvf ../lsfiles ../findfiles
  ```
  The output should be empty.
* build again.

File: zcutil/distclean.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+# Copyright (c) 2020 The Zcash developers
+
+zcutil/clean.sh
+
+rm -rf depends/*-*-*
+rm -rf depends/work
+rm -rf depends/built
+rm -rf depends/sources
+rm -rf target
```
Comment (str4d) [inline zcutil/distclean.sh:10]: Move this to `clean.sh`.

File: zcutil/distclean.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+# Copyright (c) 2020 The Zcash developers
+
+zcutil/clean.sh
+
+rm -rf depends/*-*-*
+rm -rf depends/work
```
Comment (str4d) [inline zcutil/distclean.sh:7]: While this isn't generated as a result of running the root makefile, it is generated as a result of running a `make` command, so move this to `clean.sh`.

**Comment by daira** [comment]
I made the changes @str4d suggested, and checked that this still satisfies the intent that
```
zcutil/distclean.sh
zcutil/build.sh
zcutil/clean.sh
zcutil/build.sh
```
does not rebuild dependencies on the second invocation of `build.sh`.

Please rereview.

**Comment by str4d** [comment]
@zkbot r+

**Pull request merged**